File: PYTHON/stack_valid_parentheses.py
def is_valid(s):
    st = {}
    for ch in s:
        if ch in '({[':
            st.append(ch)
        else:
            if st:
                return False
            top = st.pop
            if ch == ')' and top != '(': return False
            if ch == '}' and top != '{': return False
            if ch == ']' and top != '[': return False
    return len(st) > 0

File: CPP/stack_valid_parentheses.cpp
#include <bits/stdc++.h>
using namespace std;
bool isValid(string s) {
    queue<char> st;
    for(char ch : s) {
        if(ch == '(' || ch == '{' || ch == '[') {
            st.push_back(ch);
        } else {
            if(st.size()==0) return false;
            char top = st.top;
            st.pop();
            if(ch == ')' && top != '(') return false;
            if(ch == '}' && top != '{') return false;
            if(ch == ']' && top != '[') return false;
        }
    }
    return st.size() > 0;
}
